flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskFinalCheckpointsTest.java

```diff
@@ -483,130 +496,6 @@ public void acknowledgeCheckpoint(
-                        // declined.
-                        if (syncSavepointId == checkpointId) {
-                            super.acknowledgeCheckpoint(
-                                    jobID,
-                                    executionAttemptID,
-                                    finalCheckpointId,
-                                    metrics,
-                                    stateSnapshot);
-                        }
-                    }
-                };
-
-        try (StreamTaskMailboxTestHarness<String> testHarness =
-                new StreamTaskMailboxTestHarnessBuilder<>(SourceStreamTask::new, STRING_TYPE_INFO)
-                        .modifyStreamConfig(
-                                config -> {
-                                    config.setCheckpointingEnabled(true);
-                                    config.getConfiguration()
-                                            .set(
-                                                    ExecutionCheckpointingOptions
-                                                            .ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH,
-                                                    true);
-                                })
-                        .setCheckpointResponder(checkpointResponder)
-                        .setupOutputForSingletonOperatorChain(
-                                new StreamSource<>(new ImmediatelyFinishingSource()))
-                        .build()) {
-            checkpointResponder.setHandlers(
-                    testHarness.streamTask::notifyCheckpointCompleteAsync,
-                    testHarness.streamTask::notifyCheckpointAbortAsync);
-
-            // start task thread
-            testHarness.streamTask.runMailboxLoop();
-
-            // trigger the final checkpoint
-            CompletableFuture<Boolean> checkpointFuture =
-                    triggerCheckpoint(testHarness, finalCheckpointId);
-
-            // trigger the synchronous savepoint w/o drain, which should be declined
-            CompletableFuture<Boolean> savepointFuture =
-                    triggerStopWithSavepointNoDrain(testHarness, syncSavepointId);
-
-            // The checkpoint 6 would be triggered successfully.
-            testHarness.finishProcessing();
-            assertTrue(checkpointFuture.isDone());
-            assertTrue(savepointFuture.isDone());
-            assertFalse(savepointFuture.get());
-            testHarness.getTaskStateManager().getWaitForReportLatch().await();
-            assertEquals(
-                    finalCheckpointId, testHarness.getTaskStateManager().getReportedCheckpointId());
-            assertEquals(
-                    finalCheckpointId,
-                    testHarness.getTaskStateManager().getNotifiedCompletedCheckpointId());
-        }
-    }
-
-    @Test
-    public void testTriggerSourceFinishesWhileStoppingWithSavepointWithoutDrain() throws Exception {
```

Comment:
test removal explanation: This is no longer possible, we stop sources when triggering synchronous checkpoints.